cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/util/ProgramRunMetaFileUtil.java

```diff
@@ -0,0 +1,131 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.twill.filesystem.Location;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Utility class for reading and writing program run meta files.
+ */
+public class ProgramRunMetaFileUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(ProgramRunMetaFileUtil.class);
+
+  private static final Schema STARTING_INFO = Schema.recordOf(
+    "ProgramStartingInfo",
+//    Schema.Field.of("artifactName", Schema.of(Schema.Type.STRING)),
+//    Schema.Field.of("artifactVersion", Schema.of(Schema.Type.STRING)),
+//    Schema.Field.of("artifactScope", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.USER, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.RUNTIME_ARGUMENTS, Schema.mapOf(Schema.of(Schema.Type.STRING),
+                                                              Schema.of(Schema.Type.STRING)))
+    );
+
+  private static final String SCHEMA_STRING = Schema.recordOf(
+    "ReportRecord",
+    Schema.Field.of(Constants.NAMESPACE, Schema.of(Schema.Type.STRING)),
+//    Schema.Field.of("application", Schema.of(Schema.Type.STRING)),
+//    Schema.Field.of("version", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.PROGRAM, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.RUN, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.STATUS, Schema.of(Schema.Type.STRING)),
+    Schema.Field.of(Constants.TIME, Schema.of(Schema.Type.LONG)),
+    Schema.Field.of(Constants.START_INFO,  Schema.nullableOf(STARTING_INFO))
+  ).toString();
+
+  public static final String RUN_META_FILE = "/Users/Chengfeng/tmp/run_meta.avro";
+  public static final org.apache.avro.Schema STARTING_INFO_SCHEMA =
+    new org.apache.avro.Schema.Parser().parse(STARTING_INFO.toString());
+  public static final org.apache.avro.Schema SCHEMA = new org.apache.avro.Schema.Parser().parse(SCHEMA_STRING);
+
+  private static final class ProgramStartInfo {
+    private final String user;
+
+    ProgramStartInfo(String user) {
+      this.user = user;
+    }
+
+    public String getUser() {
+      return user;
+    }
+  }
+
+  public static ProgramStartInfo startingInfo(String user) {
+    return new ProgramStartInfo(user);
+  }
+
+  public static void populateMetaFiles(Location metaBaseLocation) throws Exception {
+    DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(ProgramRunMetaFileUtil.SCHEMA);
+    DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter);
+    for (String namespace : ImmutableList.of("default", "ns1", "ns2")) {
```

Comment:
can you removed hard-coded code ? similarly loop count 5 below. this method also needs javadoc and comments.